cirq-ionq/cirq_ionq/ionq_devices.py

```diff
@@ -13,94 +13,14 @@
-
-
-class IonQAPIDevice(cirq.Device):
-    """A device that uses the gates exposed by the IonQ API.
-
-    When using this device in constructing a circuit, it will convert one and two qubit gates
-    that are not supported by the API into those supported by the API if they have a unitary
-    matrix (support the unitary protocol).
-
-    Note that this device does not do any compression of the resulting circuit, i.e. it may
-    result in a series of single qubit gates that could be executed using far fewer elements.
-
-    The gates supported by the API are
-        * `cirq.XPowGate`, `cirq.YPowGate`, `cirq.ZPowGate`
-        * `cirq.XXPowGate`, `cirq.YYPowGate`, `cirq.ZZPowGate`
-        * `cirq.CNOT`, `cirq.H`, `cirq.SWAP`
-        * `cirq.MeasurementGate`
-    """
-
-    def __init__(self, qubits: Union[Sequence[cirq.LineQubit], int], atol=1e-8):
-        """Construct the device.
-
-        Args:
-            qubits: The qubits upon which this device acts or the number of qubits. If the number
-                of qubits, then the qubits will be `cirq.LineQubit`s from 0 to this number minus
-                one.
-            atol: The absolute tolerance used for gate calculations and decompositions.
-        """
-        if isinstance(qubits, int):
-            self.qubits = frozenset(cirq.LineQubit.range(qubits))
-        else:
-            self.qubits = frozenset(qubits)
-        self.atol = atol
-        self._metadata = cirq.DeviceMetadata(
-            self.qubits,
-            [(a, b) for a in self.qubits for b in self.qubits if a != b],
-        )
-
-    @property
-    def metadata(self) -> cirq.DeviceMetadata:
-        return self._metadata
-
-    @_compat.deprecated(
-        fix='Use metadata.qubit_set if applicable.',
-        deadline='v0.15',
-    )
-    def qubit_set(self) -> AbstractSet['cirq.Qid']:
-        return self.qubits
-
-    def validate_operation(self, operation: cirq.Operation):
-        if operation.gate is None:
-            raise ValueError(
-                f'IonQAPIDevice does not support operations with no gates {operation}.'
-            )
-        if not self.is_api_gate(operation):
-            raise ValueError(f'IonQAPIDevice has unsupported gate {operation.gate}.')
-        if not set(operation.qubits).intersection(self.metadata.qubit_set):
-            raise ValueError(f'Operation with qubits not on the device. Qubits: {operation.qubits}')
-
-    def is_api_gate(self, operation: cirq.Operation) -> bool:
-        return operation in _VALID_GATES
+from cirq import ops, protocols, _import
+from cirq.transformers.target_gatesets import compilation_target_gateset
 
-    @_compat.deprecated(
-        fix='Use cirq_ionq.decompose_to_device operation instead.',
-        deadline='v0.15',
-    )
-    def decompose_operation(self, operation: cirq.Operation) -> cirq.OP_TREE:
-        return decompose_to_device(operation)
+drop_empty_moments = _import.LazyLoader('drop_empty_moments', globals(), 'cirq.transformers')
```

Comment:
is there a reason to lazy load these? I think if you've loaded cirq above `import cirq` you'll pull these already since they are defined in a transitive dependency.

Reply:
Nope! This was copy-pasta only, removed it.